python/TestHarness/testers/SchemaDiff.py

```diff
@@ -0,0 +1,172 @@
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+from RunApp import RunApp
+from TestHarness import util
+import json, os
+
+class SchemaDiff(RunApp):
+    @staticmethod
+    def validParams():
+        params = RunApp.validParams()
+        params.addParam('schemadiff',   [], "A list of XML or JSON files to compare.")
+        params.addParam('gold_dir',      'gold', "The directory where the \"gold standard\" files (the expected output to compare against) reside relative to the TEST_DIR: (default: ./gold/). This only needs to be set if the gold file is in a non-standard location.")
+        params.addParam('gold_file',      None, "Specify the file in the gold_dir that the output should be compared against. This only needs to be set if the gold file uses a different file name than the output file.")
+        params.addParam('ignored_items',  [], "Items in the schema that the differ will ignore. These can be keys or values, i.e. for \"foo\": \"bar\", either foo or bar can be chosen to be selected. Note that entering a value located inside a list will skip the whole list.")
+        params.addParam('rel_err',       5.5e-6, "Relative error value allowed in comparisons. If rel_err value is set to 0, it will work on the absolute difference between the values.")
+        params.addParam('abs_zero',      1e-10, "Absolute zero cutoff used in diff comparisons. Every value smaller than this threshold will be ignored."),
+        return params
+
+    def __init__(self, name, params):
+        RunApp.__init__(self, name, params)
+        if self.specs['required_python_packages'] is None:
+            self.specs['required_python_packages'] = 'deepdiff xmltodict'
+        elif 'deepdiff' not in self.specs['required_python_packages']:
+            self.specs['required_python_packages'] += ' deepdiff'
+        elif 'xmltodict' not in self.specs['required_python_packages']:
```

Comment:
This should only be required if we can find `.xml` files in the `schemadiff` parameter